apps/notifications/src/doc/note-rendering.md

```diff
@@ -0,0 +1,131 @@
+
+These notification block components will be your entry point for any modifications you need to do to the way a particular notification is rendered to the user.
+
+For instance, if the user has received a notification about a post on a blog they are following, the component in question will be `src/panel/notifications/block-post.jsx`:
+
+```js
+import React from 'react';
+
+import { html } from '../indices-to-html';
+import { p } from './functions';
+
+const PostBlock = ( { block } ) => <div className="wpnc__post">{ p( html( block ) ) }</div>;
+
+export default PostBlock;
+```
+
+First, the block is passed to the `html()` function from `/src/panel/indices-to-html/index.js`, which processes the block's content information (making use of the `ranges` mentioned earlier) to split that information into chunks based on how they should be rendered. These chunks are then processed to reintroduce HTML tags back into the plain text content at the points indicated by the values in the `ranges` property.
+
+The result of this function (a string enhanced with HTML tag information) is then passed to `p()` in `src/panel/templates/functions.jsx`, which adds further HTML tag formatting based on line breaks and other special characters (such as backticks for code blocks).
+
+## Troubleshooting and Updating Renderings
+
+### Calypso rendering logic
+
+Local development of Calypso logic for rendering notifications can take place making use of the Calypso dev server, as any changes made to `src/apps/notifications` will be reflected in the Calypso local development environment automatically.
+
+However, as notifications can also be viewed in an iframe on non-Calypso sites using the masterbar (see https://github.com/Automattic/wp-calypso/edit/master/apps/notifications/README.md), any changes to the notifications sub-application should also be tested on a sandboxed WP.com site. To do this, run the following commands:
+
+```bash
+# Builds files and places them in `apps/notifications/dist`
+cd apps/notifications
+yarn build
+```
+
+Then copy the built `dist` folder to your sandbox environment to test updated rendering behavior.
+
+### Server rendering logic
+
+If a block is rendering incorrectly due to formatting information being lost before the data reaches Calypso, you may need to perform further troubleshooting on the server-side logic used to generate the list of a user's notifications.
+
+This server logic can be found in the WPCOM codebase, and its primary entry point is in `/wp-content/lib/class.notifications-builder.php:1046` in a call to `Notification_Block::add_post_block()`. Further information about the server logic associated with notifications can be found in the PCYsg-46P-p2 FG entry.
```

Comment:
Should this be another line number? L1046 is a `$statsd::timing()` call.

Reply:
L1046 seems correct to me (I see a `$this->add_block()` call there) but yeah, we cannot point to a specific commit revision like in GitHub, so L1046 can get deprecated if new changes are pushed to `/wp-content/lib/class.notifications-builder.php`.

Maybe we can leave it as just `/wp-content/lib/class.notifications-builder.php`.